plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomTimeEditor.java

```diff
@@ -369,4 +371,18 @@ public void selectAllContent() {
         }
     }
 
+    @Nullable
+    public Date adaptToDate(@Nullable Object value) throws ClassCastException {
+        if (value == null) {
+            return null;
+        } else if (value instanceof Date) {
+            return (Date) value;
+        } else if (value instanceof Instant) {
+            return Date.from((Instant) value);
+        } else if (value instanceof LocalDateTime) {
+            return Date.from(Instant.from((LocalDateTime) value));
+        } else {
+            throw new ClassCastException(value.toString());
```

Comment:
Well, we try to adapt the value to the required type. Value can be String and not be adaptable without having specific date transformers. So it probably should be a class cast exception, maybe IllegalArgumentException will also be ok, but in any case, ClassCastException is more specific.

Reply:
btw I found `IllegalArgumentException` more suitable too

Reply:
It's not specific for the callee as it's generally used by the Java standard library and JVM.

Catching `ClassCastException` looks like a crutch and a buggy code, not like it's intended.

Actually, I would even prefer to use `DBException` because it's checked and will force the callee to process it.

Reply:
@ShadelessFox `DBException` will be problematic to handle because it's already thrown by methods in the same part as adaptToDate which will lead to possible mixup of two causes when processing.

Reply:
Agreed, never use ClassCastException. Create special exception class if needed.